misc/diagram-util/src/main/java/org/ballerinalang/diagramutil/connector/models/connector/ReferenceType.java

```diff
@@ -34,26 +34,46 @@
 
 import java.util.HashMap;
 import java.util.Map;
+import java.util.Optional;
 
 public class ReferenceType {
     private static final Map<String, RefType> visitedTypeMap = new HashMap<>();
     
     public record Field(String fieldName, RefType type, boolean optional, String defaultValue) {
+        public RefType getType() {
+            return type;
+        }
     }
 
     public static RefType fromSemanticSymbol(Symbol symbol) {
         SymbolKind kind = symbol.kind();
-        TypeSymbol typeSymbol = null;
+        TypeSymbol typeSymbol;
+        String name = "";
         if (kind == SymbolKind.TYPE_DEFINITION) {
             typeSymbol = ((TypeDefinitionSymbol) symbol).typeDescriptor();
+            name = symbol.getName().orElseThrow();
         } else if (kind == SymbolKind.PARAMETER) {
             typeSymbol = ((ParameterSymbol) symbol).typeDescriptor();
+            name = typeSymbol.getName().orElseThrow();
```

Comment:
The type-descriptor returning from `typeDescriptor()` is not always have a name (e.x. `record {| string id; |} person`, here `person` param's type doesn't have a name because its not a type reference). Hence I don't think throwing an error is appropriate.